### DISCUSSION TITLE
blog/why-react-isnt-dying

### DISCUSSION THREAD

**Discussion by giscus** [TkDodo/blog-comments#92] [Announcements]
# blog/why-react-isnt-dying

Finishing my train of thought about why React is here to stay (for now).

https://tkdodo.eu/blog/why-react-isnt-dying#inject-comments

**Comment by Moshyfawn** [comment]
While it's funny to me seeing "React gave us **a predictable** way to render our UI" with all the latest Suspense behaviour, fetch patching etc., in this context, I completely agree with that React experimented with "revolutionary" _(in UI dev world)_ concepts that are now been built upon by modern solutions _(how fast can calling software a "modern solutions" age with the current eco-system? 🤭)_.

If React would stop investing in R&Ds and iterate on the ways to improve right now, that's where I draw the "it's dead" line. With the current eco-system and tools its been compared to, I don't think it's happening any time soon, though.

One big concern about React "dying" I have they keep investing and breaking "the experience" by making changes "inspired" by backwards-compatibility, which you can see in how slow the classes to hooks transition was and the most recent example: the negative talk about CRA experience and how "outdated" of experience it is.

I feel like in the software field, being an innovator implies you need to sacrifices some of that 100% compatibility and avoid looking back at your past too much. Otherwise, you start to loose your momentum and that's where devs start to "loose interest" and consider you "dead".

**Comment by ahuegle** [comment]
Hm, see your article on daily.dev. I notice that more and more entries from daily.dev are pure advertisments.
What is your intension? Write aboute react ore promote https://tanstack.com ? When i checked your github repository first of all i see a forked project from https://tanstack.com. Sorry at least for me the discussion about react is dying is useless.

**Reply by Moshyfawn** [reply]
This blog post is a direct continuation of the discussion they had on [State of React Ecosystem Jan 2023](https://youtu.be/_nGuk2Gs2oc?t=2924) that got interrupted due to an outage. Not a promotion.

Dominik is a core Tanstack Query maintainer. https://tanstack.com/ is the landing page for Tanstack projects, including React Query.

**Reply by ahuegle** [reply]
Oh i see, in this context (State of React Ecosystem Jan 2023) the article makes sense of course. 
I am very impressed on Dan Abramov as redux maintainer and also as react context api maintainer wich he is now i think. 
Thx for explanation.

**Comment by Thiagolino8** [comment]
I agree with many of the points listed, but one point that I don't agree with so much is related to the ecosystem, because despite having much less specific libraries for the framework there is also a smaller need for specific libraries
One of the biggest advantages of new frameworks, especially Svelte, is that by working with web standards and directly with the DOM (instead of the VDOM used by React and Vue) any library designed to work with vanilla javascript and/or with the DOM is automatically compatible, not needing a specific wrapper for the framework (ex: i18n, motion one, auto-animate, etc...)
And more specifically in the case of Svelte, because it relies on a contract store model for its reactivity instead of a specific reactivity library like SolidJs, it is very easy to use reactive libraries that were not thought at first to work with Svelte, even libraries made in principle to work with React (ex: preact-signals, zustand, redux, valtio, rxjs)
In addition, many of the largest React libraries were created to fill gaps that React had, gaps that these frameworks either do not have or already provide an integrated solution.
Considering these points, I think it's safe to say that because they don't need framework-specific libraries as much, they actually have a larger ecosystem in many areas, the clearest exception being UI component libraries where it's considerably harder to do anything. agnostic (I believe the only option would be something done using web components), and this exception is where React has a huge advantage and will continue to do so for a long time to come

**Comment by sanneh2** [comment]
React is the best out there. The problem is with elite anti patterning that is trying to gatekeep themeselves into relevance, but actually harming to coding community. Custom hooks are a great general disparity in programming and while they have some benefits, they do not have all the benefits and could considerably be dropped and foregone. Functional components is the general best possible pattern ever invented and everything can be done inside of them using cohesion. The sooner the community understands this, the better we will all be by just relaxing and doing what already works and is 100% verified and tested.

**Reply by c01nd01r** [reply]
> Functional components is the general best possible pattern ever invented and everything can be done inside of them using cohesion. 

Conceptually, functional components are good, but not as implemented in React. Your code gets recreated and reinvoked at every turn. No currently popular framework behaves this way.
At the same time, take a look at SolidJS - it uses functional components, JSX, and true reactivity. Additionally, you can safely write code within "render functions".

**Reply by TkDodo** [reply]
there's still a trade-off in solidJs. Like, you can't use early returns in your functional components (see https://x.com/TkDodo/status/1526804959957770241). So fine grained reactivity and signals have their drawbacks / things to be aware of, too.

I stand by that for most apps, just having "your code recreated and reinvoked at every turn" is good enough, and it gives you the upside of being able to write javascript however you choose.

**Comment by OzzyTheGiant** [comment]
Having more libraries available does not mean better support. Take a look at the Angular and Svelte communities, they have almost everything they need baked in and don't need much other than UI enhancements and CSS frameworks. React is dead to me considering that it decided for the hooks and functional component anti-pattern. UI will always be best represented by objects because state has to be saved to memory anyways. The more I repeat this, the easier it will become over time for people to start catching on and dropping React.

**Comment by Chriscaracach** [comment]
Great post. Your "growth loop" explanation was really interesting. Thanks
